resources/views/service/centreusers/edit.blade.php

```diff
@@ -0,0 +1,147 @@
+            </div>
+            <button type="submit" id="createWorker">Add worker</button>
+        </form>
+        <script>
+            function buildCheckboxes() {
+                // Setup data for checkboxes
+                var data = $("#worker_centre")
+                    .find(':selected')
+                    .siblings()
+                    .map(function(index,centre) {
+                        return {
+                            id : centre.value,
+                            name : centre.text,
+                            selected : $(centre).attr('data-workercentre')
+                        }
+                    });
+
+                // Set the default
+                var boxes =[$('<div><p>This centre has no neighbours.</p></div>')];
+
+                // Iterate data to see if we need to change that
+                if (data.length > 0) {
+                    boxes = $.map(data, function(obj) {
+                        var div = $('<div class="checkbox-group">');
+
+                        // Make an input
+                        $('<input>')
+                            .attr({
+                                type    : 'checkbox',
+                                name    : 'alternate_centres[]',
+                                value   : obj.id,
+                                id      : 'neighbour-' +obj.id
+                            })
+                            .prop('checked', !!(obj.selected))
+                            .change(updateModel)
+                            .appendTo(div);
+
+                        // Make a label for it
+                        $('<label>').attr({
+                                for  : 'neighbour-' +obj.id
+                            })
+                            .text(obj.name)
+                            .appendTo(div);
+
+                        // Return the div
+                        return div;
+                    });
+                }
+                // build checkboxes
+                $('#centres').children().remove();
+                $.each(boxes, function(i, box){
+                    box.appendTo('#centres');
+                });
+            }
+
+            // This updates our "model", select-option elements decorates with data-* items.
+            function updateModel(e) {
+                // Find the centre equiv to the checkbox what we changed.
+                var centre = $('#worker_centre').find('option[value="' + e.target.value + '"]');
+                $(e.target).prop('checked')
+                    //if it's checked, set to 'alternate'
+                    ? centre.attr('data-workercentre', 'alternate')
+                    //if it's not checked, remove attribute
+                    : centre.removeAttr('data-workercentre')
+                ;
+                // rebuild the checkboxes.
+                buildCheckboxes();
+            }
+
+            // Start the page
+            $(document).ready(
+                function () {
```

Comment:
So sad about all the `jquery` but :man_shrugging:

Reply:
"just enough javascript"